extensions/ql-vscode/src/common/files.ts

```diff
@@ -132,3 +132,30 @@ export function isIOError(e: any): e is IOError {
 export function tmpdir(): string {
   return osTmpdir();
 }
+
+/**
+ * Finds the common parent directory of an arbitrary number of paths. If the paths are absolute,
```

Comment:
Are the input paths expected to be files or directories, or potentially either or a mixture?

I ask this because arguably `findCommonParentDir("foo/") === "foo/")` but `findCommonParentDir("foo/query.ql") === "foo/"`.

We may have to check which paths are files and call `dirname` on them.